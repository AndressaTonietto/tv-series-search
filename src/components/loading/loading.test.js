import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import Loading from "./loading";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

describe("Loading Component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <Loading />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders loading spinner", () => {
    render(<Loading />);
    const spinnerElement = screen.getByTestId("loading-spinner");
    expect(spinnerElement).toBeInTheDocument();
  });
});
