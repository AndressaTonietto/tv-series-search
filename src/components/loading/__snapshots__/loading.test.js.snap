// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component renders correctly 1`] = `
<div
  className="flex justify-center"
>
  <span
    className="loading loading-spinner loading-sm"
    data-testid="loading-spinner"
  />
</div>
`;
